crates/rrg/src/action/execute_signed_command.rs

```diff
@@ -0,0 +1,568 @@
+// in the LICENSE file or at https://opensource.org/licenses/MIT.
+
+use std::{
+    collections::HashMap,
+    env,
+    io::{Read, Write},
+    process::{Command, ExitStatus},
+};
+
+use protobuf::Message;
+
+// TODO(s-westphal): Check and update max size.
+const MAX_OUTPUT_SIZE: usize = 4048;
+const COMMAND_EXECUTION_CHECK_INTERVAL: std::time::Duration = std::time::Duration::from_secs(1);
+
+/// Arguments of the `execute_signed_command` action.
+pub struct Args {
+    raw_command: Vec<u8>,
+    command: rrg_proto::execute_signed_command::SignedCommand,
+    stdin: Stdin,
+    ed25519_signature: ed25519_dalek::Signature,
+    timeout: std::time::Duration,
+}
+
+/// Result of the `execute_signed_command` action.
+pub struct Item {
+    /// Exit status of the command subprocess.
+    exit_status: ExitStatus,
+    /// Standard output of the command executiom.
+    stdout: Vec<u8>,
+    /// Wheather standard output is truncated.
+    truncated_stdout: bool,
+    /// Standard error of the command executiom.
+    stderr: Vec<u8>,
+    /// Wheather stderr is truncated.
+    truncated_stderr: bool,
+}
+
+enum Stdin {
+    None,
+    Unsigned(Vec<u8>),
+    Signed(Vec<u8>),
+}
+
+/// An error indicating that the command signature is missing.
+#[derive(Debug)]
+struct MissingCommandSignatureError;
+
+impl std::fmt::Display for MissingCommandSignatureError {
+    fn fmt(&self, fmt: &mut std::fmt::Formatter) -> std::fmt::Result {
+        write! {
+            fmt,
+            "missing command signature"
+        }
+    }
+}
+
+impl std::error::Error for MissingCommandSignatureError {}
+
+/// An error indicating that stdin of the command couln't be captured.
+#[derive(Debug)]
+struct CommandStdinError;
+
+impl std::fmt::Display for CommandStdinError {
+    fn fmt(&self, fmt: &mut std::fmt::Formatter) -> std::fmt::Result {
+        write! {
+            fmt,
+            "failed to capture stdin when spawning the command process"
+        }
+    }
+}
+
+impl std::error::Error for CommandStdinError {}
+
+/// Handles invocations of the `execute_signed_command` action.
+pub fn handle<S>(session: &mut S, mut args: Args) -> crate::session::Result<()>
+where
+    S: crate::session::Session,
+{
+    match session.args().command_verification_key {
+        Some(key) => key
+            .verify_strict(&args.raw_command, &args.ed25519_signature)
+            .map_err(crate::session::Error::action)?,
+        None => return Err(crate::session::Error::action(MissingCommandSignatureError)),
+    };
+
+    let command_path = &std::path::PathBuf::try_from(args.command.take_path())
+        .map_err(crate::session::Error::action)?;
+
+    // // For calling `cmd /C <command>` in windows, C:\Windows\System32
+    // // needs to be on the PATH, otherwise <command> is not found.
+    // let filtered_env: HashMap<String, String> = env::vars()
+    //     .filter(|&(ref k, _)| k.to_lowercase() == "path")
+    //     .collect();
```

Comment:
Let's remove this commented out part (and the `.envs(filtered_env)` part below).

Reply:
Done.